docs/java/features/extensions/extension-library/curconv/currency-conversion.md

```diff
@@ -13,69 +13,225 @@ keywords:
+
+This is done by using an exponent value of the target currency. An exponent represents the relationship between a major and minor currency unit as a mathematical function. For example, 100 cents make one US Dollar (USD). We can represent this relationship by using a "base value" and an "exponent value". For all rounding off calculations, the base value is fixed as 10. Therefore, in this example, the exponent will be 2.
+
+Additionally, converted amounts in the library response are rounded off to a "scale" equal to the exponent of the target currency. This means that if the source currency's exponent value is 2 (as is the case in our example), there will be two digits after the decimal in the rounded off amount.
+
+Exponents for all currencies are published by ISO and this information is also available in the [Currency](https://docs.oracle.com/javase/7/docs/api/java/util/Currency.html) class.
+
+This rounding off logic is applied to all conversion amounts.
+
+:::note
+The library provides both the unrounded and rounded-off amounts in the result set. You can choose which one to opt for, based on your use case.
+:::
+
+## Features
+
+### Convert Currencies
+Get currency exchange rates converted from a base currency to a target currency by using the Currency Conversion library.
+
 #### Get Bulk Conversions
 Get multiple currency pairs converted through a single call. You can use the library methods to get more than one currency pair converted at the same time.
 
-```java
-//example code
-```
-
 #### Get Direct and Indirect Conversions
 Get direct and indirect currency conversions for your currency pairs. Direct conversions are one-to-one conversions between a source and a target currency. Indirect conversions are performed by using inverse rates or a reference currency.
 
+### Multi-tenancy with Currency Conversion
+Based on the dependency added to pom.xml, the library identifies the environment you are operating on and the tenant context. The library automatically derives the tenant context by using the [Tenant abstraction](https://help.sap.com/doc/19838225037e4e52ba36d2bb95176126/1.0/en-US/com/sap/cloud/sdk/cloudplatform/tenant/Tenant.html) provided by the SAP Cloud SDK. To override the default conversion configuration, manually add an exchange rate provider and an exchange rate provider code to the API request, as follows. If you do this, the default configuration settings are not taken into consideration.
+
 ```java
-//example code
-```
+TenantAccessor.executeWithTenant(providerTenant, CallableWhichPerformsCurrencyConversion);
 
-#### Use Integration Objects
-Use integration objects that contain data models and UIs, and services that operate on the data models that you can use to send information to the Currency Conversion library.
+TenantAccessor.executeWithTenant(() -> MyProviderTenant, () -> {
+  // Call the Currency Conversion library here.
+});
+```
+## Code Example
```

Comment:
Divide this section into subsection or each use case instead of showing one big code snippet

Reply:
The above code snippet isn't "usable code" but rather an illustrative example for how the process flow on the whole should look. We feel sub-sections might introduce redundancies in code, which we'd like to avoid. For example, library and parameter initializations. 

We would like to retain this as is.